pkg/instance/service.go

```diff
@@ -361,3 +339,54 @@ func (s service) FindInstances(user *model.User, presets bool) ([]GroupWithInsta
 
 	return instances, err
 }
+
+func (s service) Reset(token string, instance *model.Instance) error {
+	decrypted, err := s.FindByIdDecrypted(instance.ID)
+	if err != nil {
+		return errdef.NewNotFound(err)
+	}
+
+	err = s.destroy(token, decrypted)
+	if err != nil {
+		return err
+	}
+
+	err = s.Deploy(token, decrypted)
+	if err != nil {
+		return err
```

Comment:
What should a user do if deploying again fails? I think we should return that as part of the error 😄 Since we are not retrying to deploy for them.

Reply:
But what should a user do?